main/src/ca/uwaterloo/flix/language/phase/PatternExhaustiveness.scala

```diff
@@ -678,6 +683,7 @@ object PatternExhaustiveness {
     case Pattern.Cst(Ast.Constant.Int64(_), _, _) => TyCon.Int64
     case Pattern.Cst(Ast.Constant.BigInt(_), _, _) => TyCon.BigInt
     case Pattern.Cst(Ast.Constant.Str(_), _, _) => TyCon.Str
+    case Pattern.Cst(Ast.Constant.Regex(_), _, _) => TyCon.Regex
```

Comment:
See other comment about weeding out Regex literals in match cases. This case should be unreachable (as with Null below) and we shouldn't need this TyCon.

Reply:
I've made the `Ast.Constant.Regex(_),..` case throw an error like the Null case below.